### PR TITLE
Fix new logins with Simplified SS using the proxy

### PR DESCRIPTION
**Pull request by jmartinesp**
## Content

Setup whether to use the proxy or native sliding sync at an earlier stage when signing in, so the `SlidingSyncVersion` is properly populated, stored and later restored.

## Motivation and context

With the current code, Simplified Sliding Sync was never used even if it was enabled in developer options.

## Tests

- Enable SSS.
- Check the logs and ensure the `/sync` requests hit the right endpoint (`_matrix/client/unstable/org.matrix.simplified_msc3575/sync` instead of `_matrix/client/unstable/org.matrix.msc3575/sync`).
- Disable SSS.
- Check again the endpoint for the `/sync` requests.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 14

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClientFactory.kt [hunk 1]
```diff
@@ -95,11 +90,18 @@ class RustMatrixClientFactory @Inject constructor(
...
+    internal suspend fun getBaseClientBuilder(
         sessionPaths: SessionPaths,
         passphrase: String?,
-        slidingSync: ClientBuilderSlidingSync,
+        restore: Boolean,
     ): ClientBuilder {
+        val slidingSync = when {
+            AuthenticationConfig.SLIDING_SYNC_PROXY_URL != null -> ClientBuilderSlidingSync.CustomProxy(AuthenticationConfig.SLIDING_SYNC_PROXY_URL!!)
+            appPreferencesStore.isSimplifiedSlidingSyncEnabledFlow().first() -> ClientBuilderSlidingSync.Simplified
+            restore -> ClientBuilderSlidingSync.Restored
```
Comment (ganfra) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClientFactory.kt:101]: Should we not checked first if restore is `true`?

Comment (jmartinesp) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClientFactory.kt:101]: Ooops, true.

Comment (jmartinesp) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClientFactory.kt]: https://github.com/element-hq/element-x-android/pull/3417/commits/ae3f555fa081f2a6497185c9eeb9b50d0456c70d

**Review by ganfra — APPROVED** [review]
Thanks for the fix, I've not tested but looks ok!

**Pull request merged**